modules/nextflow/src/main/groovy/nextflow/file/FileCollector.groovy

```diff
@@ -237,20 +239,28 @@ abstract class FileCollector implements Closeable {
 
         // verify if a cached list exists
         final hash = makeHash()?.toString()
-        Path temp = hash ? FileHelper.getLocalTempPath().resolve("${hash}.collect-file") : null
```

Comment:
My concern is that temp path was set on the local disk on purpose because the `collectFile` sorts the collect items using using the local file system. I wonder how this can work using moving it over the pipeline work dir when using a remote path such as `s3://something`

Reply:
Great point. I'll see if we can fix this by writing locally and then moving into workDir. Failing that, perhaps this list of filenames can be stored in the db.

Reply:
Without any changes, this works when using S3 as a workdir. 

I tested this by running Nextflow on an EC2 VM and using the awsbatch executor, running the workflow twice:

![image](https://user-images.githubusercontent.com/31576/217956589-c65b568d-bc5f-4488-995b-3360897313fc.png)

The collectfile cache is written as expected to `${workDir}/tmp/<hash>/<hash>/collectFile.cache`:

![image](https://user-images.githubusercontent.com/31576/217956974-3600b12a-757e-40a6-8c74-1b92cea5ac0f.png)

Reply:
I believe you, but HOW? 😄

Reply:
Rob, I think Paolo is asking about how this change affects the sorting of the files.

My understanding is that this meta file only contains the list of paths but not the contents. On AWS Batch, the chunk files will still be sorted on local scratch storage, whereas the meta file will be generated in memory and written once to the work dir. So sorting performance should not be affected.

Reply:
Yup, you're right Ben. 

The file in question (the one I'm calling `collectFile.cache`) does not store the sorted contents, but is a serialized list of paths.